File: apps/flutter_app/lib/features/chat/domain/entities/member.dart
import 'package:equatable/equatable.dart';

enum MemberRole {
  owner,
  admin,
  moderator,
  member,
}

class Member extends Equatable {
  final String userId;
  final String displayName;
  final String? avatarUrl;
  final MemberRole role;
  final DateTime joinedAt;
  final bool isOnline;

  const Member({
    required this.userId,
    required this.displayName,
    this.avatarUrl,
    this.role = MemberRole.member,
    required this.joinedAt,
    this.isOnline = false,
  });

  @override
  List<Object?> get props => [
        userId,
        displayName,
        avatarUrl,
        role,
        joinedAt,
        isOnline,
      ];
}

File: web-client/src/domain/entities/Member.ts
import type { User } from './User';

export type MemberRole = 'owner' | 'admin' | 'member';

export interface Member {
  id: string;
  userId: string;
  roomId: string;
  user: User;
  role: MemberRole;
  joinedAt: Date;
}
